src/store/plugins/external-layers.plugin.js

```diff
@@ -21,73 +20,98 @@ const dispatcher = { dispatcher: 'external-layers.plugin' }
+            ...dispatcher,
+        })
     }
     store.subscribe((mutation, state) => {
         if (mutation.type === 'addLayer') {
-            addLayerSubscriber(mutation.payload.layer, state)
+            layersSubscriber([mutation.payload.layer], state)
         }
         if (mutation.type === 'setLayers') {
-            mutation.payload.layers?.forEach((layer) => {
-                addLayerSubscriber(layer, state)
-            })
+            layersSubscriber(mutation.payload.layers, state)
         }
     })
 }
 
-async function updateExternalLayer(store, externalLayer, projection) {
-    try {
-        let updatedExternalLayer = null
-        if (
-            externalLayer instanceof ExternalWMSLayer ||
-            externalLayer instanceof ExternalGroupOfLayers
-        ) {
-            updatedExternalLayer = await updatedWMSLayerAttributes(externalLayer, projection)
-        } else if (externalLayer instanceof ExternalWMTSLayer) {
-            updatedExternalLayer = await updatedWMTSLayerAttributes(externalLayer, projection)
-        } else {
-            throw new Error(`Unsupported type of layer: ${typeof externalLayer}`)
-        }
+function getWMSCababilitiesForLayers(layers) {
+    const capabilities = {}
+    // here we use a Set to take the unique URL to avoid loading multiple times the get capabilities
+    // for example when adding several layers from the same source.
+    new Set(
+        layers
+            .filter(
+                (layer) =>
+                    layer instanceof ExternalWMSLayer || layer instanceof ExternalGroupOfLayers
+            )
+            .map((layer) => layer.baseUrl)
+    ).forEach((url) => {
+        capabilities[url] = readWmsCapabilities(url)
+    })
+    return capabilities
+}
 
-        updatedExternalLayer.isLoading = false
-        store.dispatch('updateLayer', {
-            layer: updatedExternalLayer,
-            ...dispatcher,
-        })
+function getWMTSCababilitiesForLayers(layers) {
+    const capabilities = {}
+    // here we use a Set to take the unique URL to avoid loading multiple times the get capabilities
+    // for example when adding several layers from the same source.
+    new Set(
+        layers.filter((layer) => layer instanceof ExternalWMTSLayer).map((layer) => layer.baseUrl)
+    ).forEach((url) => {
+        capabilities[url] = readWmtsCapabilities(url)
+    })
+    return capabilities
+}
+
+async function updateExternalLayer(store, capabilities, layer, projection) {
+    try {
+        const resolvedCapabilities = await capabilities
+        log.debug(
+            `Update External layer ${layer.id} with capabilities`,
+            layer,
+            resolvedCapabilities
+        )
+        const updated = resolvedCapabilities.getExternalLayerObject(
+            layer.externalLayerId,
+            projection,
+            layer.opacity,
+            layer.visible,
+            false /* throw Error in case of  error */
+        )
```

Comment:
I think this is out of scope of this PR. Although I agree that it would be better I'd like to concentrate on the go live instead of doing code refactoring when not necessary.